src/open_inwoner/openklant/models.py

```diff
@@ -208,8 +209,12 @@ def api_url(self):
         return urljoin(self.api_root, self.api_path)
 
     @classmethod
-    def from_django_settings(cls):
+    def from_django_settings(cls) -> Self:
         from django.conf import settings
 
-        if config := getattr(settings, "OPENKLANT2_CONFIG", None):
-            return cls(**config)
+        if not (config := getattr(settings, "OPENKLANT2_CONFIG", None)):
+            raise RuntimeError(
```

Comment:
This makes branching on eSuite/OpenKlant2 a bit more tedious. We then need an additional flag to check if one or the other should be used; otherwise tests fail with this `RuntimeEror`. See my PR link below. Since this is provisional, not a big thing

https://github.com/maykinmedia/open-inwoner/pull/1474/files

Reply:
Yeah, `ImproperlyConfigured` sound more on target.